### PR TITLE
Move storage tests to use the framework/log package

### PR DESCRIPTION
**Pull request by johnSchnake**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This is part of the transition to using framework/log instead
of the Logf inside the framework package. This will help with
import size/cycles when importing the framework or subpackages.

**Which issue(s) this PR fixes**:
Ref #77359

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig testing
/area test
/area test-framework

### CODE REVIEW

File: test/e2e/storage/regional_pd.go [hunk 1]
```diff
@@ -189,19 +190,19 @@ func testZonalFailover(c clientset.Interface, ns string) {
...
-		framework.Logf("deleting claims in namespace %s", ns)
+		e2elog.Logf("deleting claims in namespace %s", ns)
 		pvc := getPVC(c, ns, regionalPDLabels)
 		framework.ExpectNoError(c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Delete(pvc.Name, nil),
 			"Error deleting claim %s.", pvc.Name)
 		if pvc.Spec.VolumeName != "" {
 			err = framework.WaitForPersistentVolumeDeleted(c, pvc.Spec.VolumeName, framework.Poll, pvDeletionTimeout)
 			if err != nil {
-				framework.Logf("WARNING: PV %s is not yet deleted, and subsequent tests may be affected.", pvc.Spec.VolumeName)
+				e2elog.Logf("WARNING: PV %s is not yet deleted, and subsequent tests may be affected.", pvc.Spec.VolumeName)
```
Comment (andrewsykim) [inline test/e2e/storage/regional_pd.go]: @draveness another usecase where exposing the log level in `e2elog` could be useful.

**Review by andrewsykim — COMMENTED** [review]
/lgtm

**Review by timothysc — APPROVED** [review]
/approve

**Pull request merged**